### PR TITLE
Gradle version bump and other changes

### PR DESCRIPTION
**Pull request by MG-LSJ**
Changes:

- Gradle version updated to 7.4
- Nearby connections library updated (Unsure if it has any major breaking changes)
- Example app library bumps
- Some minor changes

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,37 +1,53 @@
...
+    compileSdk 34
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion 19
```
Comment (mannprerak2) [inline android/build.gradle:37]: With this we probably should increase the major version in pubspec.yaml.

Comment (mannprerak2) [reply android/build.gradle:37]: Also please add to changelog.yml about these changes

File: android/build.gradle [hunk 2]
```diff
@@ -1,37 +1,53 @@
...
-        classpath 'com.android.tools.build:gradle:4.1.2'
+        classpath 'com.android.tools.build:gradle:7.4.2'
     }
 }
 
 rootProject.allprojects {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
```
Comment (mannprerak2) [inline android/build.gradle:18]: Any particular reason for changing this? It seens like jcenter is a superset of mavencentral

File: android/build.gradle [hunk 3]
```diff
@@ -1,37 +1,53 @@
 group 'com.pkmnapps.nearby_connections'
-version '1.0-SNAPSHOT'
+version '1.0'
```
Comment (mannprerak2) [inline android/build.gradle:2]: I think the convention is to keep -SNAPSHOT so lets keep this as it is.

File: android/.project [hunk 4]
```diff
@@ -1,34 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<projectDescription>
-	<name>nearby_connections</name>
-	<comment>Project android created by Buildship.</comment>
-	<projects>
-	</projects>
-	<buildSpec>
-		<buildCommand>
-			<name>org.eclipse.jdt.core.javabuilder</name>
```
Comment (mannprerak2) [inline android/.project:9]: Why are files in .settings, .project, . classpath deleted?

I think they are supposed to be kept in version control?

File: README.md [hunk 5]
```diff
@@ -2,107 +2,124 @@
...
     Permission.bluetoothConnect,
     Permission.bluetoothScan
 ].request();
+
+// Check Bluetooth Status
+await Permission.bluetooth.serviceStatus.isEnabled;
+
+
+// Android 12+
+await Permission.nearbyWifiDevices.request()
```
Comment (mannprerak2) [inline README.md:100]: Great job on the readme updates 👍

**Review by mannprerak2 — CHANGES_REQUESTED** [review]
Thank you for submitting this 😃. Hopefully we can land this soon.

File: android/build.gradle (hunk 2 above)
Comment (MG-LSJ) [reply android/build.gradle:18]: https://developer.android.com/build/jcenter-migration

File: android/.project (hunk 4 above)
Comment (MG-LSJ) [reply android/.project:9]: I created a new flutter plugin project with the latest settings, and then ported over your java code. These seems to not exist in the the new project.

Comment (mannprerak2) [reply android/.project:9]: cool

**Review by mannprerak2 — APPROVED** [review]

**Comment by mannprerak2** [comment]
Thanks @MG-LSJ 🙌🏻 🎉 

The new version is also live on pub.dev.

**Pull request merged**